hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -222,19 +222,6 @@ def _check_company_id_add_line_task_id(self):
                     _('The Company in the Timesheet Sheet and in '
                       'the Task must be the same.'))
 
-    @api.constrains('company_id')
-    def _check_company_id(self):
-        for rec in self.sudo():
```

Comment:
why is obsolete this constrain?

Reply:
That constraint was dead code, I was not able to trigger it functionally, nor by executing tests.
Field `company_id` is [readonly](https://github.com/OCA/hr-timesheet/blob/12.0/hr_timesheet_sheet/models/hr_timesheet_sheet.py#L133).

The `_check_timesheet_sheet_company_id()` should cover what `def _check_company_id()` was supposed to do.